crates/rpc/rpc/src/eth/api/mod.rs

```diff
@@ -238,4 +257,6 @@ struct EthApiInner<Client, Pool, Network> {
     signers: Vec<Box<dyn EthSigner>>,
     /// The async cache frontend for eth related data
     eth_cache: EthStateCache,
+    /// The async gas oracle frontend for gas price suggestions
+    gas_oracle: GasPriceOracle<Client>,
```

Comment:
this is fine, 
I don't think we need more flexibility here since gas sampling will be similar for all networks?
if not we can easily replace this with a trait and box this.